### DISCUSSION TITLE
can not include header file

### DISCUSSION THREAD

**Discussion by linshuxi** [xmake-io/xmake#2734] [Q&A]
Hi,thank you very much for the great work. However, i get some problem and any suggestion is appreciated.

I am trying to build the openspiel library. The header files are at different folders, and the source files include the header files based on the relative path. for example, in an source file, it tries to include a header file by the line "open_spiel/abseil-cpp/absl/random/bit_gen_ref.h". My xmake.lua is:
target("test")
    set_kind("binary")
    add_includedirs("open_spiel/")
    add_includedirs("open_spiel/abseil-cpp/absl/random")
    add_files("open_spiel/spiel.cc")

However, neither command "add_includedirs("open_spiel/")" nor the command "add_includedirs("open_spiel/abseil-cpp/absl/random")" works.

Could you give me any suggestions?

Thanks

**Comment by waruqi** [comment]
you can add `-v` to xmake build command, and check verbose cflags (-Ixxx)

**Reply by linshuxi** [reply]
Hi, waruqi, thanks a lot for the quick response! 
The output is as follows, and the output information "-Iopen_spiel" suggests that open_spiel is correct, however, another output information "fatal error: open_spiel/abseil-cpp/absl/random/bit_gen_ref.h" suggests that gcc cannot find it. Could you give more information on that?


(base) xiao@xiao-System-Product-Name:~/projects/my_spiel/open_spiel-master$ xmake -v
checking for gcc ... /usr/bin/gcc
checking for dmd ... no
checking for ldc2 ... no
checking for gdc ... no
checking for zig ... no
checking for zig ... no
checking for gcc ... /usr/bin/gcc
checking for the c++ compiler (cxx) ... gcc
checking for /usr/bin/gcc ... ok
checking for flags (-fPIC) ... ok
[ 62%]: cache compiling.release open_spiel/spiel.cc
/usr/bin/gcc -c -m64 -Iopen_spiel -o build/.objs/test/linux/x86_64/release/open_spiel/spiel.cc.o open_spiel/spiel.cc
checking for flags (-MMD -MF) ... ok
checking for flags (-fdiagnostics-color=always) ... ok
error: In file included from open_spiel/spiel.cc:15:
open_spiel/spiel.h:30:10: fatal error: open_spiel/abseil-cpp/absl/random/bit_gen_ref.h: 没有那个文件或目录
   30 | #include "open_spiel/abseil-cpp/absl/random/bit_gen_ref.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

**Reply by waruqi** [reply]
you added incorrect includedirs.

try 
```lua
add_includedirs(".")
```

**Reply by linshuxi** [reply]
By the way, the folder information is as follows:
![2022-08-30 09-17-05屏幕截图](https://user-images.githubusercontent.com/6730777/187326542-13876dee-fdd0-4c9f-ab23-e2a37c222555.png)

**Reply by linshuxi** [reply]
thanks a lot, it works!
